### PR TITLE
[cloud_init] Update the plugin

### PR DESCRIPTION
**Pull request by slashdd**
* Snap can found be found as a DEB or SNAP

* Dropping /var/lib/cloud as it may reveal
sensible data

* Integrate commands inspired by
cloud-init collect-logs

Closes: #2080

Signed-off-by: Eric Desrochers <eric.desrochers@canonical.com>

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [x] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [x] Is the subject and message clear and concise?
- [x] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [x] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [x] If this commit closes an existing issue, is the line `Closes: #ISSUENUMBER` included in an independent line?
- [x] If this commit resolves an existing pull request, is the line `Resolves: #PRNUMBER` included in an independent line?

### CODE REVIEW

File: sos/report/plugins/cloud_init.py [hunk 1]
```diff
@@ -18,10 +18,15 @@ class CloudInit(Plugin, RedHatPlugin, UbuntuPlugin, DebianPlugin):
     plugin_name = 'cloud_init'
     packages = ('cloud-init',)
     services = ('cloud-init',)
+    commands = ('cloud-init',)
 
     def setup(self):
+        self.add_cmd_output([
+            'cloud-init --version',
+            'journalctl --boot=0 -o short-precise'
```
Comment (TurboTurtle) [inline sos/report/plugins/cloud_init.py:26]: I'm not sure a full `journal` collection makes sense here - shouldn't this be limited to the `cloud-init` unit?

**Comment by BryanQuigley** [comment]
what is an example of the sensitive data that /var/lib/cloud was capturing and can we just filter specific bad bits?

**Comment by slashdd** [comment]
> what is an example of the sensitive data that /var/lib/cloud was capturing and can we just filter specific bad bits?
```

/var/lib/cloud/instance/user-data.txt

  --include-userdata, -u
                        Optionally include user-data from
                        /var/lib/cloud/instance/user-data.txt which could
                        contain sensitive information.
```

Additionnally, I notice cloud-init collect-logs doesn't even collect `/var/lib/cloud `anymore.

**Comment by BryanQuigley** [comment]
Collecting /var/lib/cloud was requested rather recently - in #1681.    Need to dig deeper into the data  there.

**Comment by BryanQuigley** [comment]
There is also /run/cloud-init/instance-data-sensitive.json,  which on my install is hte same as the non-sensitive version. 
And these partially completed docs..  https://cloudinit.readthedocs.io/en/latest/topics/dir_layout.html

The user-data is the cloud config.. which could have a password in it, but is also really helpful to see IMHO.  Not sure the best way to address.

**Comment by slashdd** [comment]
We could use `do_file_sub()` for the password found in `user-data.txt`

For the entire contents of `/var/lib/cloud` even cloud-init `cloudinit/cmd/devel/logs.py` doesn't collect it which is the upstream way to collect data for debug purposes.

It only focuses on `/var/lib/cloud/instance/user-data.txt` iff `--include-userdata` is used.

**Pull request closed without merging**